pkg/skaffold/generate_pipeline/profile_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generatepipeline
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestGenerateProfile(t *testing.T) {
+	var tests = []struct {
+		description     string
+		skaffoldConfig  *latest.SkaffoldConfig
+		expectedProfile *latest.Profile
+		responses       []string
+		shouldErr       bool
+	}{
+		{
+			description: "successful profile generation",
```

Comment:
nice! you could also add a unit test without artifacts to make sure the error is handled as expected